file-system/directory.js

```diff
@@ -1,9 +1,57 @@
 module.exports = Directory
 
 function Directory(name) {
-
+  this.name = name
+  this._ls = []
+  this._ls_la = []
+  this._cat = ''
 }
 
 Directory.prototype.ls = function() {
+  return this._ls.map(f_data => f_data.file).reverse()
+}
+
+Directory.prototype.ls_la = function() {
+  return this._ls_la.reverse()
```

Comment:
Try remove the `this._ls_la` variable. You can implement this with something like `this._ls.map()`